### ISSUE TITLE
urllib.unquote causes a StackOverflowException

### ISSUE THREAD

**Issue by slozier** [IronLanguages/main#1414]
`urllib.unquote` causes a `StackOverflowException`.

```
import urlparse
urlparse.unquote("%20")
```

**Comment by simplicbe** [comment]
`_is_unicode` seems to cause this issue. And `_is_unicode` is defined here: 

```python
try:
    unicode
except NameError:
    def _is_unicode(x):
        return 0
else:
    def _is_unicode(x):
        return isinstance(x, unicode)
```

`_is_unicode` is calling `return isinstance(x, unicode)`

**Comment by simplicbe** [comment]
Here is the problem, that caus the stackoverflow exception:

![differencepythonipy](https://cloud.githubusercontent.com/assets/14359329/18564882/da86abe8-7b8d-11e6-810f-da216164e5ca.PNG)

It should be '$20' not '%20' which cause the exception.

**Comment by simplicbe** [comment]
This is the issue: https://github.com/IronLanguages/main/issues/1415

**Comment by simplicbe** [comment]
Now I've tracked down the real issue :)

`_is_unicode` always returns true, so `unquote` is calling it self. 

`_is_unicode` is defined here:

``` python
try:
    unicode
except NameError:
    def _is_unicode(x):
        return 0
else:
    def _is_unicode(x):
        return isinstance(x, unicode) # <-- This one is taken
```

When attaching to debugger, `unicode` is of type `str`. So that is the problem.

**Comment by slide** [comment]
Closed via #1625
